addon/-private/system/references/has-many.js

```diff
@@ -121,6 +300,39 @@ HasManyReference.prototype._isLoaded = function() {
   });
 };
 
+/**
+   `value()` sycronously returns the current value of the belongs-to
+    relationship. Unlike `record.get('relationshipName')`, calling
+    `value()` on a reference does not trigger a fetch if the async
+    relationship is not yet loaded. If the relationship is not loaded
+    it will always return `null`.
+
+   Example
+
+   ```javascript
+   // models/post.js
+   export default DS.Model.extend({
+     comments: DS.hasMany({ async: true })
+   });
+
+   var post = store.push({
+     type: 'post',
+     id: 1,
+     relationships: {
+       comments: {
+         data: [{ type: 'comment', id: 1 }]
+       }
+     }
+   });
+
+   var commentsRef = post.hasMany('comments');
+
+   commentsRef.value() === post.get('comments')
```

Comment:
This code sample is not completely accurate. We should either make the relationship sync, or resolve the `post.get("comments")` and state that the resolved value is `commentsRef.value()`.

I would prefer the promise version to indicate that references allow you synchronous access to an async relationship.